custom/icds_reports/migrations/sql_templates/database_functions/child_health_monthly.sql

```diff
@@ -0,0 +1,69 @@
+CREATE OR REPLACE FUNCTION create_child_health_monthly_view() RETURNS VOID AS
+$BODY$
+DECLARE
+  _ucr_child_list_table text;
+BEGIN
+  EXECUTE format('SELECT table_name FROM ucr_table_name_mapping WHERE table_type = %s', '''child_list''') INTO _ucr_child_list_table;
+
+  EXECUTE 'DROP VIEW IF EXISTS child_health_monthly_view CASCADE';
+  EXECUTE format('CREATE VIEW child_health_monthly_view AS ' ||
+    'SELECT' ||
+        '"child_list".case_id, ' ||
+        '"child_list".awc_id, ' ||
+        '"child_list".supervisor_id, ' ||
+        '"child_list".block_id, ' ||
+        '"child_list".district_id, ' ||
+        '"child_list".state_id, ' ||
+        '"child_list".name AS person_name, ' ||
+        '"child_list".mother_name, ' ||
+        '"child_list".dob, ' ||
+        '"child_list".sex, ' ||
+        'child_health_monthly.month, ' ||
+        'child_health_monthly.age_in_months, ' ||
+        'child_health_monthly.open_in_month, ' ||
+        'child_health_monthly.valid_in_month, ' ||
+        'child_health_monthly.nutrition_status_last_recorded, ' ||
+        'child_health_monthly.current_month_nutrition_status, ' ||
+        'child_health_monthly.pse_days_attended, ' ||
+        'child_health_monthly.recorded_weight, ' ||
+        'child_health_monthly.recorded_height, ' ||
+        'child_health_monthly.current_month_stunting, ' ||
+        'child_health_monthly.current_month_wasting, ' ||
+        'child_health_monthly.thr_eligible, ' ||
+        'GREATEST(child_health_monthly.fully_immunized_on_time, child_health_monthly.fully_immunized_late) AS fully_immunized, ' ||
+        'CASE WHEN child_health_monthly.current_month_nutrition_status = %2$s THEN 1' ||
+            'WHEN child_health_monthly.current_month_nutrition_status = %3$s THEN 2' ||
+            'WHEN child_health_monthly.current_month_nutrition_status = %6$s THEN 3' ||
+            'ELSE 4 END AS current_month_nutrition_status_sort,' ||
+        'CASE WHEN child_health_monthly.current_month_stunting = %4$s THEN 1' ||
+            'WHEN child_health_monthly.current_month_stunting = %5$s THEN 2' ||
+            'WHEN child_health_monthly.current_month_stunting = %6$s THEN 3' ||
+            'ELSE 4 END AS current_month_stunting_sort,' ||
+        'CASE WHEN child_health_monthly.current_month_wasting = %4$s THEN 1' ||
+            'WHEN child_health_monthly.current_month_wasting = %5$s THEN 2' ||
+            'WHEN child_health_monthly.current_month_wasting = %6$s THEN 3' ||
+            'ELSE 4 END AS current_month_wasting_sort,' ||
+        'CASE ' ||
+          'WHEN child_health_monthly.zscore_grading_hfa = 1 AND child_health_monthly.zscore_grading_hfa_recorded_in_month = 1 THEN %4$s ' ||
```

Comment:
Is the `%4$s` indexing into an array that's fed into this? I don't think I've seen this type of syntax before

Reply:
@emord the ```format``` function which I used as the first parameter takes the string and next parameters are arguments, So ```%4$s``` means that this will be replaced by 4th argument and this argument must be a string.

for Example:
```
    format('some %1$s string, and %2$s text', 'awesome', 'more')
```
the result will be:
```
    some awesome string, and more text
```

When I tried to remove all ```quote_ident``` and ```quote_literals``` I found only this possibility.